src/utils/beacon/useHandleBeaconMessage.test.tsx

```diff
@@ -0,0 +1,378 @@
+      network: { type: NetworkType.MAINNET },
+      appMetadata: { name: "mockDappName", senderId: "mockSenderId" },
+      sourceAddress: account.address.pkh,
+    };
+
+    const {
+      result: { current: handleMessage },
+    } = renderHook(useHandleBeaconMessage);
+
+    act(() => handleMessage(message));
+
+    await waitFor(() =>
+      expect(dynamicModalContextMock.openWith).toHaveBeenCalledWith(
+        <BeaconSignPage
+          fee={BigNumber(100)}
+          message={message}
+          operation={
+            makeAccountOperations(account, account, [mockTezOperation(1)]) as ImplicitOperations
+          }
+        />
+      )
+    );
+    expect(mockToast).not.toHaveBeenCalled();
+  });
+
+  it("opens a modal with the BatchSignPage for multiple operations", async () => {
+    mockEstimatedFee(100);
+
+    const message: BeaconRequestOutputMessage = {
+      type: BeaconMessageType.OperationRequest,
+      operationDetails: [
+        {
+          kind: TezosOperationType.TRANSACTION,
+          amount: "1",
+          destination: mockImplicitAddress(2).pkh,
+        },
+        {
+          kind: TezosOperationType.TRANSACTION,
+          amount: "1",
+          destination: mockImplicitAddress(2).pkh,
+        },
+      ],
+      senderId: "mockSenderId",
+      id: "mockMessageId",
+      network: { type: NetworkType.MAINNET },
+      appMetadata: { name: "mockDappName", senderId: "mockSenderId" },
+      sourceAddress: account.address.pkh,
+    };
+
+    const {
+      result: { current: handleMessage },
+    } = renderHook(useHandleBeaconMessage);
+
+    act(() => handleMessage(message));
+
+    await waitFor(() =>
+      expect(dynamicModalContextMock.openWith).toHaveBeenCalledWith(
+        <BatchSignPage
+          fee={BigNumber(100)}
+          message={message}
+          operation={
+            makeAccountOperations(account, account, [
+              mockTezOperation(1),
+              mockTezOperation(1),
+            ]) as ImplicitOperations
+          }
+        />
+      )
+    );
+    expect(mockToast).not.toHaveBeenCalled();
+  });
+});
+
+describe("toAccountOperations", () => {
+  it("throws if the list is empty", () => {
+    expect(() => toAccountOperations([], account)).toThrow("Empty operation details!");
+  });
+
+  it("converts a list of partial operations to ImplicitOperations", () => {
+    const operationDetails: PartialTezosOperation[] = [
+      {
+        kind: TezosOperationType.TRANSACTION,
+        amount: "1",
+        destination: mockImplicitAddress(2).pkh,
+      },
+      {
+        kind: TezosOperationType.TRANSACTION,
+        amount: "2",
+        destination: mockImplicitAddress(2).pkh,
+      },
+    ];
+
+    const operations = toAccountOperations(operationDetails, account);
+
+    expect(operations).toEqual(
+      makeAccountOperations(account, account, [
+        mockTezOperation(1),
+        { ...mockTezOperation(2), recipient: mockImplicitAddress(2) },
```

Comment:
Can we update `mockTezOperation` or create another mock instead of using it like this? 
It's not very clear what's going on

Reply:
I inlined the values here because changing mockTezOperation is too much for this PR. but I agree it should be updated